src/pages/ContactInfoPage.tsx

```diff
@@ -0,0 +1,247 @@
+  DialogContentText,
+  DialogTitle,
+  IconButton,
+  Button,
+  Snackbar,
+} from "@mui/material";
+import ArrowBackIcon from "@mui/icons-material/ArrowBack";
+import {
+  Contact,
+  ContactEventChannel,
+  ContactService,
+} from "@adorsys-gis/contact-service";
+import { eventBus } from "@adorsys-gis/event-bus";
+import {
+  ServiceResponse,
+  ServiceResponseStatus,
+} from "@adorsys-gis/status-service";
+import { MessageRepository } from "../storage/MessageRepository";
+
+const messageRepository = new MessageRepository();
+
+const ContactInfoPage: React.FC = () => {
+  const { contactId } = useParams<{ contactId: string }>();
+  const navigate = useNavigate();
+  const [contactName, setContactName] = useState<string>("");
+  const [contactDID, setContactDID] = useState<string>("");
+  const [openDialog, setOpenDialog] = useState(false);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarMessage, setSnackbarMessage] = useState("");
+
+  const contactService = new ContactService(eventBus);
+
+  // Helper function to get initials from the contact's name
+  const getInitials = (name: string) => {
+    const words = name.trim().split(" ");
+    return words.length > 1
+      ? `${words[0][0].toUpperCase()}${words[1][0].toUpperCase()}`
+      : words[0][0].toUpperCase();
+  };
+
+  useEffect(() => {
+    if (contactId) {
+      const id = parseInt(contactId);
+      contactService.getContact(id);
+
+      const handleContactReceived = (response: ServiceResponse<Contact>) => {
+        if (
+          response.status === ServiceResponseStatus.Success &&
+          response.payload
+        ) {
+          setContactName(response.payload.name);
+          setContactDID(response.payload.did || "");
+        } else {
+          console.error(response.payload);
+        }
+      };
+
+      const getContactChannel = ContactEventChannel.GetContactByID;
+      eventBus.on(getContactChannel, handleContactReceived);
+
+      return () => {
+        eventBus.off(getContactChannel, handleContactReceived);
+      };
+    } else {
+      console.error("Contact ID is undefined.");
+    }
+  }, [contactId, contactService, eventBus]);
+
+  // Opens delete confirmation dialog
+  const handleDeleteClick = () => {
+    setOpenDialog(true);
+  };
+
+  // Closes the delete confirmation dialog
+  const handleCloseDialog = () => {
+    setOpenDialog(false);
+  };
+
+  // Handles deletion of contact
+  const handleConfirmDelete = () => {
+    if (contactId) {
+      const id = parseInt(contactId);
+      contactService.deleteContact(id);
+
+      const handleDeleteResponse = (
+        response: ServiceResponse<{ id: number }>
+      ) => {
+        if (response.status === ServiceResponseStatus.Success) {
+
+          // Delete all messages associated with this contact
+          messageRepository.deleteAllByContact(contactId);
+
+          setSnackbarMessage("Contact and messages removed successfully."); // Set success message
```

Comment:
Instead of using the Snackbar directly, you can use our own status-library and event-bus library for handling notifications.